TACC/Core-Portal
# task/FP-369: DataFiles Toolbar Compress and Extract buttons

## Overview: ##

Compress & Extract Buttons in Data Files

## PR Status: ##


## Related Jira tickets: ##

* [FP-369](https://jira.tacc.utexas.edu/browse/FP-369)

## Summary of Changes: ##

- Compress using zippy-frontera app and Extract using extract-frontera apps
- These buttons will bring up modals with some UI (@owaisj please confirm these match UI-Design's specs)
- Added onCancel support so SystemPushKeysModal
- If these apps require pushing keys, they will bring up the SystemPushKeysModal. If pushing keys was canceled, they will show an error icon in the compress or extract actions buttons, as this is required

## Testing Steps: ##

- Start ngrok for notifications
- Remove your frontera.fork.CLI authorized_key from your frontera.tacc.utexas.edu:~/.ssh/authorized_keys file
- Try compress and extract, with cancellation of push keys modal (should not allow you to proceed)
- Try compress and extract, but push keys this time to make sure it executes
- Try it again to make sure you are not prompted for new keys since those have already been pushed
- Add  `"compressApp": 'zippy-frontera', "extractApp": 'extract-frontera'` to `_WORKBENCH_SETTINGS`

## UI Photos:

> As of 2477c1e (Merge: ecc54cf6 87ac2c31)
- __Compress - Modal (Success)__
  ![FP-369 Compress Modal](https://user-images.githubusercontent.com/62723358/102656481-8ed1fd80-4139-11eb-94e0-faa3fdbc92de.png)
- __Expand - Modal (Success)__
  ![FP-369 Expand Modal](https://user-images.githubusercontent.com/62723358/102656486-90032a80-4139-11eb-95c7-ca2e79baf26f.png)
- __Compress - Job Entry__
  ![FP-369 Compress Job Entry](https://user-images.githubusercontent.com/62723358/102658016-5f70c000-413c-11eb-8283-90836cf54917.png)
- __Compress - Job Details__
  ![FP-369 Compress Job Details](https://user-images.githubusercontent.com/62723358/102658021-613a8380-413c-11eb-991c-a08ecd759b71.png)

## Notes: ##

- extract-frontera can probably exposed as with metadata or apps metadata replacement for 
[FP-687](https://jira.tacc.utexas.edu/browse/FP-687)
